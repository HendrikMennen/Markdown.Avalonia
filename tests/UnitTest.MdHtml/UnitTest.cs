using ApprovalTests;
using ApprovalTests.Reporters;
using Avalonia.Controls;
using Markdown.Avalonia;
using Markdown.Avalonia.Html;
using NUnit.Framework;
using System.Linq;
using UnitTest.Base;
using UnitTest.Base.Utils;
using UnitTest.MdHtml.Test;

using MarkdownEngine = Markdown.Avalonia.Markdown;

namespace UnitTest.MdHtml
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTest : UnitTestBase
    {
        public MarkdownEngine Engine
        {
            get
            {
                var plugins = new MdAvPlugins();
                plugins.Plugins.Add(new HtmlPlugin());

                return new MarkdownEngine()
                {
                    Plugins = plugins
                };
            }
        }

        [Test]
        [RunOnUI]
        public void Button()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void CodeBlock()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void InlineCode()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Input()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void List()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Progres()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Details()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void TypicalBlock()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void TypicalInline()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }

        [Test]
        [RunOnUI]
        public void Mixing()
        {
            var html = Utils.ReadHtml();

            var doc = Engine.Transform(html);

            var xaml = Utils.AsXaml(doc);

            Approvals.Verify(xaml);
        }
    }
}
