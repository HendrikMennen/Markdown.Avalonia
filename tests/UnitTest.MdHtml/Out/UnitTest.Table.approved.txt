<?xml version="1.0" encoding="utf-16"?>
<StackPanel xmlns:c="clr-namespace:ColorTextBlock.Avalonia;assembly=ColorTextBlock.Avalonia" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns="https://github.com/avaloniaui">
  <Border Classes="Table" xmlns="">
    <Border.Child>
      <Grid Classes="Table">
        <Border Classes="OddTableRow" Grid.Column="0" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="6">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="alpha" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="1" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="5">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="beta" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="4">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="gamma" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="3" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Classes="EvenTableRow" Grid.Column="3" Grid.Row="1" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="1" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="3" Grid.Row="2" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="2" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="3" Grid.Row="3" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="3" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="2" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="3" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="1" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="2" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="3" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Grid.ColumnDefinitions>
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
        </Grid.ColumnDefinitions>
        <Grid.RowDefinitions>
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
        </Grid.RowDefinitions>
      </Grid>
    </Border.Child>
  </Border>
  <Border Classes="Table" xmlns="">
    <Border.Child>
      <Grid Classes="Table">
        <Border Classes="OddTableRow" Grid.Column="0" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="6">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="alpha" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="1" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="5">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="beta" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="4">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="gamma" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="3" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Classes="EvenTableRow" Grid.Column="3" Grid.Row="1" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="1" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="3" Grid.Row="2" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="2" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="3" Grid.Row="3" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="3" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="4" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="3" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="1" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="2" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="3" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Grid.ColumnDefinitions>
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
        </Grid.ColumnDefinitions>
        <Grid.RowDefinitions>
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
        </Grid.RowDefinitions>
      </Grid>
    </Border.Child>
  </Border>
  <Border Classes="Table" xmlns="">
    <Border.Child>
      <Grid Classes="Table">
        <Border Classes="OddTableRow" Grid.Column="0" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="6">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="alpha" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="1" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="5">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="beta" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="4">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="gamma" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="3" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Classes="EvenTableRow" Grid.Column="3" Grid.Row="1" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="1" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="3" Grid.Row="2" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="2" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="3" Grid.Row="3" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="3" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="4" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="3" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Classes="EvenTableRow" Grid.Column="1" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="5" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="2" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="3" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Grid.ColumnDefinitions>
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
        </Grid.ColumnDefinitions>
        <Grid.RowDefinitions>
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
        </Grid.RowDefinitions>
      </Grid>
    </Border.Child>
  </Border>
  <Border Classes="Table" xmlns="">
    <Border.Child>
      <Grid Classes="Table">
        <Border Classes="OddTableRow" Grid.Column="0" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="6">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="alpha" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="1" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="5">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="beta" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="4">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="gamma" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="3" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Classes="EvenTableRow" Grid.Column="3" Grid.Row="1" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="1" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="3" Grid.Row="2" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="2" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="3" Grid.Row="3" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="3" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="4" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="3" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Classes="EvenTableRow" Grid.Column="1" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="5" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="2" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="3" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Classes="OddTableRow" Grid.Column="0" Grid.Row="6" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="6" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="1" Grid.Row="6" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="2" Grid.Row="6" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="3" Grid.Row="6" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Grid.ColumnDefinitions>
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
        </Grid.ColumnDefinitions>
        <Grid.RowDefinitions>
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
        </Grid.RowDefinitions>
      </Grid>
    </Border.Child>
  </Border>
  <Border Classes="Table" xmlns="">
    <Border.Child>
      <Grid Classes="Table">
        <Border Classes="OddTableRow" Grid.Column="0" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="6">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="alpha" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="1" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="beta" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="gamma" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="3" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="3">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="delta" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="4" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="epsilon" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="1" Grid.Row="1" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="1" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="2" Grid.Row="1" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="2" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="4" Grid.Row="1" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Classes="OddTableRow" Grid.Column="1" Grid.Row="2" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="3" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="2" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="4" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="4" Grid.Row="2" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="5" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="1" Grid.Row="3" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="4" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="2" Grid.Row="3" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="3" Grid.Row="3" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="4" Grid.Row="3" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="1" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="2" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="3" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="4" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="1" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="2" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="3" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="4" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Grid.ColumnDefinitions>
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
        </Grid.ColumnDefinitions>
        <Grid.RowDefinitions>
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
        </Grid.RowDefinitions>
      </Grid>
    </Border.Child>
  </Border>
  <Border Classes="Table" xmlns="">
    <Border.Child>
      <Grid Classes="Table">
        <Border Classes="OddTableRow" Grid.Column="0" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="6">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="alpha" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="1" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="beta" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="gamma" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="3" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="delta" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="4" Grid.Row="0" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="epsilon" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="1" Grid.Row="1" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="1" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="2" Grid.Row="1" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="2" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="3" Grid.Row="1" Grid.ColumnSpan="1" Grid.RowSpan="3">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="D" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="4" Grid.Row="1" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Classes="OddTableRow" Grid.Column="1" Grid.Row="2" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="3" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="2" Grid.Row="2" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="4" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="OddTableRow" Grid.Column="4" Grid.Row="2" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="5" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Classes="EvenTableRow" Grid.Column="1" Grid.Row="3" Grid.ColumnSpan="1" Grid.RowSpan="1">
          <Border.Child>
            <c:CTextBlock>
              <c:CRun Text="4" />
            </c:CTextBlock>
          </Border.Child>
        </Border>
        <Border Grid.Column="3" Grid.Row="3" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="4" Grid.Row="3" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="1" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="2" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="3" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="4" Grid.Row="4" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="1" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="2" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="3" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Border Grid.Column="4" Grid.Row="5" Grid.ColumnSpan="1" Grid.RowSpan="1" />
        <Grid.ColumnDefinitions>
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
          <ColumnDefinition />
        </Grid.ColumnDefinitions>
        <Grid.RowDefinitions>
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
          <RowDefinition />
        </Grid.RowDefinitions>
      </Grid>
    </Border.Child>
  </Border>
</StackPanel>
